### PR TITLE
[XPU] Fix the bug of LoRA logits on the XPU platform

### PR DESCRIPTION
**Pull request by chaojun-zhang**
## Purpose
Fix the bug of LoRA logits on the XPU platform 
## Test Plan
VLLM_ALLOW_LONG_MAX_MODEL_LEN=1 VLLM_WORKER_MULTIPROC_METHOD=spawn python3 examples/offline_inference/multilora_inference.py

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

**Review by jeejeelee — APPROVED** [review]
The changes are relatively localized. I assume you have already verified it on XPU.

**Review by jikunshang — APPROVED** [review]
